src/CMakeLists.txt

```diff
@@ -1,13 +1,7 @@
-cmake_minimum_required(VERSION 3.0)
-
-# The library
 add_library(example complex.cpp)
 target_include_directories(example PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+# Setting this to the library only avoids errors in 3rd party dependencies which are compile with e.g. -Werror
```

Comment:
Oh yes. Solutions depend on the project setup and purpose. Supplying a general case is not easy.
In my PR I left the compile flags because they were already there, leaving them in what I thought the least intrusive way.
This is fine.